contracts/Collector.sol

```diff
@@ -9,18 +9,21 @@ import "@openzeppelin/contracts/math/SafeMath.sol";
 contract Collector is ICollector{
     IERC20 public token;
     address public owner;
+    address private remainderAddress;
     RevenuePartner[] private partners;
 
     constructor(
         address _owner,
         IERC20 _token,
-        RevenuePartner[] memory _partners
+        RevenuePartner[] memory _partners,
+        address _remainderAddress
     )
     public
     validShares(_partners)
     {   
         owner = _owner;
         token = _token;
+        remainderAddress = _remainderAddress;
         for (uint i = 0; i < _partners.length; i++)
             partners.push(_partners[i]);
```

Comment:
so here the constructor iterates over the array again after doing the same job when validating shares.

Reply:
I think it's out of the scope of this task and we should create a new task for this.

Reply:
Created a [task](https://rsklabs.atlassian.net/browse/PP-351) for this.